### PR TITLE
Multistage check

### LINKED ISSUE #839: Rework of the multistage splitting pass
since the implementation of the new lowering creates one multistage per vertical region, there is no further need for stage multistage splitting in the current sense.
This would need to be reworked to split to reduce memory pressure / halo size of individual kernel calls and should only run later down the line. It is no longer required to generate legal parallel code

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Technical Description

Adds verification check to ensure that multistages do not exceed max halo size as this would require a halo exchange.

### Resolves / Enhances

Resolves #839

### CODE REVIEW

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/test/integration-test/dawn4py-tests/tridiagonal_solve_stencil.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def main(args: argparse.Namespace):
         sir_utils.pprint(sir)
 
     # compile
-    code = dawn4py.compile(sir, backend="cuda")
+    code = dawn4py.compile(sir, backend="cuda", stage_merger=False)
```
Comment (twicki) [inline dawn/test/integration-test/dawn4py-tests/tridiagonal_solve_stencil.py:161]: awesome that this works now!

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
The changes in this PR are covered by #1010 so closing.

**Pull request closed without merging**